### PR TITLE
Feature/tvfocusguideview android

### PR DESCRIPTION
**Pull request by SebastienChauvin**
## Summary

Add support for TVFocusGuideView on Android: https://github.com/react-native-tvos/react-native-tvos/issues/206

## Changelog

Android Added - Support for TVFocusGuideView

## Test Plan

Tested with rn-tester / TVFocusGuideView

https://user-images.githubusercontent.com/7773037/173526216-7d1199c7-d532-4182-b551-9bcc23a540f4.mov

### CODE REVIEW

File: Libraries/Components/AppleTV/TVFocusGuideView.js [hunk 1]
```diff
@@ -25,42 +26,64 @@ type FocusGuideProps = $ReadOnly<{
 }>;
 
 const TVFocusGuideView = (props: FocusGuideProps) => {
-  const focusGuideRef = React.useRef(null);
+  const tvOSFocusGuideRef = React.useRef(null);
+  const [nativeDestinations, setNativeDestinations] = React.useState([]);
 
   React.useEffect(() => {
-    const nativeDestinations = (props.destinations || [])
```
Comment (douglowder) [inline Libraries/Components/AppleTV/TVFocusGuideView.js:31]: It looks like your changes are based on an earlier version of this repo. In particular, your changes would revert https://github.com/react-native-tvos/react-native-tvos/commit/6896ad6d08cc2a2bd237c9486e0f5702ad03ffea and break tvOS.

**Review by douglowder — CHANGES_REQUESTED** [review]
See inline comment

File: Libraries/Components/AppleTV/TVFocusGuideView.js (hunk 1 above)
Comment (SebastienChauvin) [reply Libraries/Components/AppleTV/TVFocusGuideView.js:31]: In my tests, tvOS was still working. I will check again the code changes.

Comment (SebastienChauvin) [reply Libraries/Components/AppleTV/TVFocusGuideView.js:31]: After looking at this again, the behaviour has not changed for tvOS. the state has been an introduced as an intermediary but ultimately Commands.setDestination is called in the same way as before.

Comment (douglowder) [reply Libraries/Components/AppleTV/TVFocusGuideView.js:31]: Have you tested your changes when running in the new Fabric architecture?

Comment (SebastienChauvin) [reply Libraries/Components/AppleTV/TVFocusGuideView.js:31]: Yes  
```
pod install          
Building RNTester with Fabric enabled.
```
--> Same behaviour
And again, the state is only there to share code between Android and tvOS. But the tvOS does the same native command (see the second useEffect)

Comment (douglowder) [reply Libraries/Components/AppleTV/TVFocusGuideView.js:31]: Thank you!!! I do see that things are working fine in local testing, so I'll go ahead and merge this.

**Review by douglowder — APPROVED** [review]

**Pull request merged**